src/core/src/pass/constant_folding.cpp

```diff
@@ -116,6 +117,37 @@ bool ov::pass::ConstantFolding::run_on_model(const std::shared_ptr<ov::Model>& m
             node->validate_and_infer_types();
         }
 
+        if (m_byte_threshold >= 0 && !node->input_values().empty() && !ov::as_type_ptr<ov::op::Sink>(node) &&
+            !ov::as_type_ptr<ov::op::v0::Result>(node)) {
+            bool is_threshold_applicable = true;
+            int64_t byte_size_before = 0;
+            for (const auto& input_value : node->input_values()) {
+                auto constant = ov::as_type_ptr<ov::op::v0::Constant>(input_value.get_node_shared_ptr());
+                if (constant && is_threshold_applicable) {
+                    byte_size_before += static_cast<int64_t>(constant->get_byte_size());
+                } else {
+                    is_threshold_applicable = false;
+                }
+            }
+
+            int64_t byte_size_after = 0;
+            for (const auto& output : node->outputs()) {
+                if (output.get_partial_shape().is_static() && is_threshold_applicable) {
+                    byte_size_after += static_cast<int64_t>(
+                        element::get_memory_size(output.get_element_type(), ov::shape_size(output.get_shape())));
+                } else {
+                    is_threshold_applicable = false;
+                }
+            }
+
+            if (is_threshold_applicable) {
+                if ((byte_size_after - byte_size_before) >= m_byte_threshold) {
+                    // do not apply ConstantFolding for this particular node because
+                    // the created Constant might increase the bin size beyond the required threshold.
+                    continue;
+                }
```

Comment:
Could you please clearify. CF_BYTE_THRESHOLD - is difference for entire model or for particular node? I see current calculation is doing for each node.